projects/packages/connection/legacy/class-jetpack-xmlrpc-server.php

```diff
@@ -326,7 +326,8 @@ public function remote_register( $request ) {
 
 			// This code mostly copied from Jetpack::admin_page_load.
 			Jetpack::maybe_set_version_option();
-			$registered = $this->connection->try_registration();
+			$from       = isset( $request['from'] ) ? $request['from'] : false;
```

Comment:
This seems reasonable. Then, @Automattic/jetpack-infinity would just need to update the Jetpack Start endpoint on the WP.com side to add the `from` arg perhaps.

Reply:
Thanks for looking at it! I've done what you suggested in https://github.com/Automattic/jetpack/pull/19609/commits/43c1cd5884bd7a159df524eb85d03ffcaa1dff30

I'm gonna try to write test instructions for all the cases and ping you if I need more help!